arroyo/processing/processor.py

```diff
@@ -177,15 +212,23 @@ def _run_once(self) -> None:
             # Otherwise, we need to try fetch a new message from the consumer,
             # even if there is no active assignment and/or processing strategy.
             try:
+                start_poll = time.time()
                 self.__message = self.__consumer.poll(timeout=1.0)
+                self.__metrics_buffer.add_poll_time(time.time() - start_poll)
```

Comment:
What about making it a context manager like we do in Sentry ? 
https://github.com/getsentry/sentry/blob/6099b7d1f450ff594e8e651896a1d574c23bf451/src/sentry/utils/snuba.py#L278-L284

It would save a statement only, this is true, but in this method I think we should get everything we can out of the way as it is already complicated enough. So the context manager could help a bit readability.

Reply:
I think this is still relevant